src/featurelib.cpp

```diff
@@ -1,4 +1,49 @@
-#include "featurelib.h"
+#include "FeatureLib.h"
 
-featurelib::featurelib() {
+FeatureLib::FeatureLib(const std::vector<std::uint8_t> &inputVector, int Width, int Height) {
+  Image = VectorToMat(inputVector, Height, Width);
+  cv::cvtColor( Image, Image, cv::COLOR_BGR2GRAY );
 }
+
+cv::Mat FeatureLib::VectorToMat(const std::vector<std::uint8_t> &inputVector, int Height, int Width) {
+  return cv::Mat(Height, Width, CV_8UC1, (std::uint8_t *) inputVector.data());
+}
+
+std::vector<uint8_t> FeatureLib::MatToVector(const cv::Mat &mat) {
```

Comment:
Check the mat to vector conversion I did in my project [here](https://github.com/TimPushkin/SgbmDepthAndroid/blob/c2b5e96da5d2fbf49c605e4233c5c8f0318ef606/sgbmdepth/src/main/cpp/src/Conversions.h#L9) and see if its suitable for you. I'm not sure which one is better, so compare them first.

Reply:
Some problems may arise if you place it in a source instead of a header file because my function is a template

Reply:
Yep, my bad
yep, i should handle with templates and abstract methods and static methods.
Tonight, i think